fix memory leaks

TaeZStkyoht opened the pull request:
- BIGNUM must be freed both on tpm2tss_tpm2data_write and tpm2tss_tpm2data_read:
definitely lost: 24 bytes in 1 blocks
indirectly lost: 8 bytes in 1 blocks
- TSSPRIVKEY's parent, privkey and pubkey members don't need to be allocated explicitly, because TSSPRIVKEY_new already does:
definitely lost: 72 bytes in 3 blocks
indirectly lost: 0 bytes in 0 blocks
- TPM2_DATA* returned from RSA_get_app_data can be returned directly genkey_rsa() because RSA_free() doesn't free this item:
definitely lost: 2,248 bytes in 1 blocks
indirectly lost: 0 bytes in 0 blocks
- update .gitignore to cover more generated files

williamcroberts commented:
You have four different tasks that were mentioned in the commit message, one for each of the list items. I think this should be broken up into four commits instead of one, so please break this up into four commits.

TaeZStkyoht committed:
tpm2tss_tpm2data_write, tpm2tss_tpm2data_read: fix memory leak

- BIGNUM must be freed both on tpm2tss_tpm2data_write and tpm2tss_tpm2data_read:
definitely lost: 24 bytes in 1 blocks
indirectly lost: 8 bytes in 1 blocks

Signed-off-by: Oguzhan Turk <stkyoht@hotmail.com>

TaeZStkyoht committed:
tpm2tss_tpm2data_write: fix memory leak

- TSSPRIVKEY's parent, privkey and pubkey members don't need to be allocated explicitly, because TSSPRIVKEY_new already does:
definitely lost: 72 bytes in 3 blocks
indirectly lost: 0 bytes in 0 blocks

Signed-off-by: Oguzhan Turk <stkyoht@hotmail.com>

TaeZStkyoht committed:
genkey_rsa: fix memory leak

- TPM2_DATA* returned from RSA_get_app_data can be returned directly genkey_rsa() because RSA_free() doesn't free this item:
definitely lost: 2,248 bytes in 1 blocks
indirectly lost: 0 bytes in 0 blocks

Signed-off-by: Oguzhan Turk <stkyoht@hotmail.com>

TaeZStkyoht committed:
update .gitignore to cover more generated files

Signed-off-by: Oguzhan Turk <stkyoht@hotmail.com>

TaeZStkyoht commented:
> You have four different tasks that were mentioned in the commit message, one for each of the list items. I think this should be broken up into four commits instead of one, so please break this up into four commits.

done

TaeZStkyoht commented:
I don't know about those errors on pipelines.

williamcroberts commented:
> I don't know about those errors on pipelines.

I don't think it's you, I need to fix this
